src/index.js

```diff
@@ -150,11 +140,11 @@ class UglifyJsPlugin {
 
   apply(compiler) {
     const requestShortener = new RequestShortener(compiler.context);
-    // Copy uglify options
-    const uglifyOptions = UglifyJsPlugin.buildDefaultUglifyOptions(this.uglifyOptions);
-    // Making sure output options exists if there is an extractComments options
-    if (this.options.extractComments) {
-      uglifyOptions.output = uglifyOptions.output || {};
+    const { uglifyOptions } = this.options;
+    const uglifiedAssets = new WeakSet();
+
+    if (typeof this.options.sourceMap === 'undefined' && (compiler.options.devtool === 'sourcemap' || compiler.options.devtool === 'source-map')) {
```

Comment:
@hulkish let's wait what other will say, but be good have separate PR for this, it is allow to maintenance CHANGELOG